app/services/delete_invoice.service.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+/**
+ * @module DeleteInvoiceService
+ */
+
+const Boom = require('@hapi/boom')
+
+const { BillRunModel, InvoiceModel } = require('../models')
+const { raw } = require('../models/base.model')
+
+class DeleteInvoiceService {
+  /**
+   * Deletes an invoice along with its licences and transactions, and updates the figures of the bill run that the
+   * invoice belongs to.
+   *
+   * @param {string} invoiceId The id of the invoice to be deleted.
+   */
+  static async go (invoiceId) {
```

Comment:
I think we also need to pass in the bill run ID so we can perform an additional security check.

We want to make sure that the invoice to be deleted is linked to the bill run ID provided. This is because thanks to the bill run plugin

- we know the bill run id exists
- we know it's linked to the regime set in the request
- we know the authenticated client system has the authority to access that regime

If we don't check the invoice is linked to the bill run, a client system could theoretically pass us any invoice ID. As long as they also provided a valid regime and bill run ID the service would delete it, even though it's not linked to this bill run.

I hope that makes sense 😕 😁

Reply:
This now uses the shiny new `FetchAndValidateBillRunInvoiceService` to check that the invoice is linked to the provided bill run.